core/templates/domain/exploration/ExplorationObjectFactory.ts

```diff
@@ -52,8 +52,8 @@ import { ExplorationChange } from './exploration-draft.model';
 
 export interface ExplorationBackendDict {
   'auto_tts_enabled'?: boolean;
-  'draft_changes': ExplorationChange[];
-  'is_version_of_draft_valid': boolean;
+  'draft_changes'?: ExplorationChange[];
+  'is_version_of_draft_valid'?: boolean;
```

Comment:
because in old code, an object of interface ReadOnlyExplorationBackendDict was passed where the object of ExplorationBackendDict is expected. When I tried to resolve this, I found that the read only exploration backend dict type was correct for the callstack below this function and exploration backend dict was correct for the callstack above that function. Both types were used with alteast 5 other functions. So, if i try to change type of parameter of one functions, then 10 more would show up for wrong data type and so-on. So, this was a issue with the typesystem. As, passing readonly exploration dict type object in place of exploration backend dict was working in the old ajs code, I made both the interfaces compatible with each other to fix the issue.

Reply:
Can we fix the test data to support the expected type instead of making params optional?

Reply:
done